app/src/main/java/com/wallet/crypto/trustapp/ui/ImportKeystoreFragment.java

```diff
@@ -39,24 +56,127 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
+                    }
+                } else {
+                    performFileSelection();
+                }
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+
+        switch (requestCode) {
+            case REQUEST_CODE_FILE_SELECTOR:
+                if (resultCode == Activity.RESULT_OK) {
+                    try {
+                        doGetSelectedKeystoreFileInfo(data);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                } else {
+                    Toast.makeText(getActivity(), getString(R.string.msg_failed_open_file_selector), Toast.LENGTH_SHORT).show();
+                }
+
+                break;
+            default:
+                break;
+        }
+
+        super.onActivityResult(requestCode, resultCode, data);
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        if (requestCode == REQUEST_CODE_PERMISSION_READ_FILE) {
+            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                performFileSelection();
+            } else {
+                // Permission Denied
+                Toast.makeText(getActivity(), "Reading Permission Denied", Toast.LENGTH_SHORT).show();
+            }
+            return;
         }
     }
 
     public void setOnImportKeystoreListener(OnImportKeystoreListener onImportKeystoreListener) {
         this.onImportKeystoreListener = onImportKeystoreListener == null
-            ? dummyOnImportKeystoreListener
-            : onImportKeystoreListener;
+                ? dummyOnImportKeystoreListener
+                : onImportKeystoreListener;
+    }
+
+    private void performFileSelection() {
+        Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        intent.setType("*/*");
+        try {
+            startActivityForResult(Intent.createChooser(intent, getString(R.string.title_select_keystore)),
+                    REQUEST_CODE_FILE_SELECTOR);
+        } catch (Exception e) {
+            Toast.makeText(getActivity(), getString(R.string.msg_failed_open_file_selector), Toast.LENGTH_SHORT).show();
+        }
+
+    }
+
+    private void doGetSelectedKeystoreFileInfo(Intent data) {
+        Uri uri = data.getData();
+        Log.d(TAG, "File Uri: " + uri.toString());
+        // Get the path
+        if (uri != null) {
+            TaskGetSelectedFileContent taskGetSelectedFileContent = new TaskGetSelectedFileContent(uri);
+            taskGetSelectedFileContent.execute();
+        }
+
+    }
+
+    private class TaskGetSelectedFileContent extends AsyncTask<Void, Void, String> {
```

Comment:
We use other approaches for background tasks.

Reply:
Thanks @madcake for your instruction.